### PR TITLE
Fix unexpected error when an empty set is returned

### LINKED ISSUE #122: Unexpected query ended
If the query has no results, the next error is thrown by the mysql client:

`ERROR 1105 (HY000): unexpected: query ended without no results and no error`

### PR DESCRIPTION
**Pull request by mcarmonaa**
Closes #122 

Signed-off-by: Manuel Carmona <manu.carmona90@gmail.com>

### CODE REVIEW

File: server/handler_test.go [hunk 1]
```diff
@@ -42,12 +42,14 @@ func TestHandlerOutput(t *testing.T) {
...
-		lastRows = len(res.Rows)
+		if res.RowsAffected != 0 {
+			lastRowsAffected = res.RowsAffected
+			lastRows = len(res.Rows)
+		}
 		return nil
 	})
 	require.NoError(err)
-	require.Equal(1, c)
+	require.Equal(2, c)
```
Comment (ajnavarro) [inline server/handler_test.go]: In this case, with a row batch of 100, it should be called only once. In this case, the second call is useless.

File: server/handler.go [hunk 2]
```diff
@@ -120,7 +122,11 @@ func (h *Handler) ComQuery(
 		r.RowsAffected++
 	}
 
-	if r != nil && r.RowsAffected != 0 {
+	// Even if r.RowsAffected = 0, the callback must be
+	// called to update the state in the go-vitess' listener
+	// an avoid to return errors when the query doesn't
```
Comment (erizocosmico) [inline server/handler.go]: s/an/and/
s/avoid to return/avoid returning/

File: server/handler.go [hunk 3]
```diff
@@ -120,7 +122,11 @@ func (h *Handler) ComQuery(
...
+	// produce results.
```
Comment (erizocosmico) [inline server/handler.go]: s/results/any results/

**Comment by mcarmonaa** [comment]
@ajnavarro ping

**Pull request merged**